lib/crossdoc/tree.rb

```diff
@@ -36,11 +36,25 @@ def download
         elsif @src.index('./')==0
           @io = open(@src.gsub('./', Dir.pwd + '/'))
         else # assume it's a URL
-          @io = URI.open(@src)
+          @io = URI.open @src
+          if @src.include?('logo_images') || num_images < 6
+            return
+          end
+          img = MiniMagick::Image.open @src
+          img.geometry image_width
```

Comment:
Not sure what all you've tried so far, but I would suggest limiting the width to the document width, converting to .jpg, and setting quality to 80% or so.

Also I think we should skip the conversion if the image is an svg since an svg in theory should be smaller than it's rasterized counterpart in most cases.

Reply:
@schultzcole - Here are some test I've just run with a Proposal containing 25 pages and 20 PNG images.
| Optimization      | PDF Size |
| ----------- | ----------- |
| None      | > 20MB       |
| Width Limit   | 12MB        |
| JPG Conversion   | 18MB        |
| Width Limit + JPG Conversion  | 2.4MB        |

Changing the image format is clearly a good move that benefits file size a lot when combined with setting the width. Thanks.

If you're in the office today, you can find some printed proposals with images on my desk where the width was limited to 1024 pixels. I haven't extensively tested it, but @ajselvig suggested keeping the pixel width higher than the document width on Monday (specifically a multiple of 2) when possible because it benefits the level of detail in the image when printed.

I agree, there's no need to apply this logic to SVGs.

Reply:
Yeah, I suppose if you're printing at 300 dpi, a page is about 2000px wide when margins are taken into account. Using 2x width seems reasonable given that fact. It'd be interesting to see what varying quality levels in the jpg encoder give us in terms of quality. I suspect that the default for image magick is around 80% which is why the pdf size was similar when you manually added the 80% quality.